src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -251,28 +266,32 @@ class DetailsCard extends React.Component {
       let fieldUpdated
       switch (fieldName) {
         case 'cluster':
-          updates = updates.set('cluster', this.props.clusters.get(value))
+          updates = updates.set('cluster', clusters.get(value))
           fieldUpdated = 'cluster'
 
           // Change the template to 'Blank' if the VM's template isn't in the new cluster
           {
-            const template = this.props.templates.get(updates.getIn(['template', 'id']))
+            const template = templates.get(updates.getIn(['template', 'id']))
             if (template && template.get('clusterId') && template.get('clusterId') !== value) {
-              changeQueue.push({ fieldName: 'template', value: this.props.blankTemplateId })
+              changeQueue.push({ fieldName: 'template', value: blankTemplateId })
             }
           }
           break
 
         case 'template':
-          updates = updates.set('template', this.props.templates.get(value))
+          updates = updates.set('template', templates.get(value))
           fieldUpdated = 'template'
 
           // Apply settings from the template to the VM (memory, CPUs, OS, cloudInit, bootMenuEnabled)
           {
-            const template = this.props.templates.get(value)
+            const template = templates.get(value)
             if (template) {
-              const templateOsType = template.getIn(['os', 'type'], 'other')
-              const templateOs = this.props.operatingSystems.find(os => os.get('type') === templateOsType)
+              const clusterId = updates.getIn(['cluster', 'id'])
+              const clusterArchitecture = getClusterArchitecture(clusterId, clusters)
+              const templateOsName = template.getIn(['os', 'type'], 'other')
```

Comment:
Why the const is **templateOsName** and not **templateOsType**?

Reply:
It's templateOsName because it's stored as `name` in the `operatingSystems` objects list:
![image](https://user-images.githubusercontent.com/64131213/102238182-2519df80-3efe-11eb-81b8-7740bd36414a.png)
Another reason, we compare it to OS Name in line [293](https://github.com/oVirt/ovirt-web-ui/blob/b63b0737f16f1354d2437560b9b55e7adb9eb92d/src/components/VmDetails/cards/DetailsCard/index.js#L293).
I'm not sure which one is clearer but if you think templateOsType is clearer I'll change that.

Reply:
The right name should really be templateOsName ("type" term is only used for templates). For clarifications I suggest adding a line comment, specifying that template.os.type and operatingSystems.name are referring to the same value on backend.